### PR TITLE
Tz size fix

### PR DESCRIPTION
**Pull request by tqiu8**
Reduce size of timezone data, by compiling zic from IANA and using `-b slim` for tz compilation. Fixes: dotnet/runtime#48061

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
What are we seeing size-wise with this change?

File: src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj [hunk 1]
```diff
@@ -35,15 +36,19 @@
 
   <Target Name="UpdateData">
     <RemoveDir Directories="$(TimeZoneDataSrcDir)" />
-    <DownloadFile SourceUrl="https://data.iana.org/time-zones/tzdb-$(TimeZoneDataVersion)/%(Area.Identity)"
-                  DestinationFolder="$(TimeZoneDataIntermediateDir)" />
+    <Exec Command="wget -r -nH --cut-dirs=2 --no-parent --reject='index.html*' -P $(TimeZoneDataIntermediateDir) https://data.iana.org/time-zones/tzdb-2021a/" />
```
Comment (eerhardt) [inline src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj:39]:
```suggestion
    <Exec Command="wget -r -nH --cut-dirs=2 --no-parent --reject='index.html*' -P $(TimeZoneDataIntermediateDir) https://data.iana.org/time-zones/tzdb-$(TimeZoneDataVersion)/" />
```

File: src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj [hunk 2]
```diff
@@ -35,15 +36,19 @@
...
+    <Exec Command="touch README" WorkingDirectory="$(TimeZoneDataIntermediateDir)"/>
```
Comment (eerhardt) [inline src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj:40]: Why is this necessary?

**Comment by lewing** [comment]
> What are we seeing size-wise with this change?

Early results are good, we're a little worried that the runtime tests aren't exhaustive enough.

**Comment by eerhardt** [comment]
> we're a little worried that the runtime tests aren't exhaustive enough.

Can you elaborate on this? Specifically what concerns do you have? In my experience, the TimeZoneInfo tests are pretty decent.

**Comment by lewing** [comment]
> > we're a little worried that the runtime tests aren't exhaustive enough.
> 
> Can you elaborate on this? Specifically what concerns do you have? In my experience, the TimeZoneInfo tests are pretty decent.

We've have absolutely run into several problems by targeting the tests too narrowly here to save size and we're just trying to avoid that again.

**Review by lewing — CHANGES_REQUESTED** [review]
This changes the build targets that aren't actually being run as part of CI, but it doesn't update the data that is generated and consumed on the dotnet/runtime side

File: src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj (hunk 2 above)
Comment (tqiu8) [reply src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj:40]: We need to download the entire tzdb directory instead of just specific files. `make zic` doesn't work without a `README`, which is not (weirdly) included in the directory itself.

**Comment by ViktorHofer** [comment]
So the binary diff is somewhere between 20-70% based on what GitHub displays. That's great to see.

**Review by ViktorHofer — APPROVED** [review]

File: src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj (hunk 2 above)
Comment (CoffeeFlux) [reply src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj:40]: ...huh, is there any chance of PRing a fix for that upstream?

Comment (lewing) [reply src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj:40]: It's possible we should switch to using the full tar.lz that contains now

**Review by lewing — APPROVED** [review]
Looks good now.  @ViktorHofer the final compressed size comes in at about 43796 bytes which is a solid savings.
